src/main/java/ch/njol/skript/conditions/CondIsBlockRedstonePowered.java

```diff
@@ -20,30 +20,60 @@
 package ch.njol.skript.conditions;
 
 import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
 import ch.njol.skript.conditions.base.PropertyCondition;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
 
 @Name("Is Block Redstone Powered")
-@Description("Checks if a block is powered by redstone")
+@Description("Checks if a block is indirectly or directly powered by redstone")
 @Examples({"if clicked block is redstone powered:",
-	"\tsend \"This block is well-powered by redstone!\""})
+	"\tsend \"This block is well-powered by redstone!\"",
+	"if clicked block is indirectly redstone powered:",
+	"\tsend \"This block is indirectly redstone powered.\""})
 @Since("2.5")
-public class CondIsBlockRedstonePowered extends PropertyCondition<Block> {
+public class CondIsBlockRedstonePowered extends Condition {
 	
 	static {
-		register(CondIsBlockRedstonePowered.class, "redstone powered", "blocks");
+		Skript.registerCondition(CondIsBlockRedstonePowered.class,
+			"%blocks% (is|are) redstone powered",
```

Comment:
Maybe we could have ``[directly] redstone powered`` as patterns as well?

Reply:
You're right; I will add it.